Mage/src/main/java/mage/abilities/effects/RedirectionEffect.java

```diff
@@ -85,6 +85,12 @@ public boolean replaceEvent(GameEvent event, Ability source, Game game) {
                 applyEffectsCounter = game.getState().getApplyEffectsCounter();
             }
         }
+        if (usageType == UsageType.ACCORDING_DURATION) {
+            amountToRedirect -= damageEvent.getAmount();
+            if (amountToRedirect <= 0) {
+                this.discard();
+            }
+        }
```

Comment:
I think this check should go down in the `restDamage > 0` check on line 105. It's already doing a check to see if all of the damage has been redirected.

Reply:
In that case I think `if restDamage > 0` should be changed to `if restDamage >= 0`, because the redirection effect should also be discarded when the dealt amount is exactly the maximum redirect amount of the effect.

However I think it makes some sense to have the UsageType-related discard calls next to each other above there.

Reply:
> In that case I think `if restDamage > 0` should be changed to `if restDamage >= 0`, because the redirection effect should also be discarded when the redirected amount is exactly the maximum redirect amount of the effect.
> 
> However I think it makes some sense to have the UsageType-related discard calls next to each other above there.

You're right. Leave it up here with the others.